doc/specs/stdlib_string_type.md

```diff
@@ -0,0 +1,1607 @@
+dlc = char(string, 1, 4)
+! dlc == "Fort"
+end
+```
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Ichar function
+
+Character-to-integer conversion function.
+
+#### Description
+
+Returns the code for the character in the first character position of the
+character sequence in the system's native character set.
+
+#### Syntax
+
+`res = [[stdlib_string_type(module):ichar(interface)]] (string)`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemental function.
+
+#### Argument
+
+- `string`: Instance of a `string_type`. This argument is `intent(in)`.
+
+#### Result value
+
+The result is a default integer scalar value.
+
+#### Example
+
+```fortran
+use stdlib_string_type
+implicit none
+type(string_type) :: string
+integer :: code
+
+string = "Fortran"
+code = ichar(string)
+end
+```
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Iachar function
+
+Code in ASCII collating sequence.
+
+#### Description
+
+Returns the code for the ASCII character in the first character position of
+the character sequences represent by the string.
+
+#### Syntax
+
+`res = [[stdlib_string_type(module):iachar(interface)]] (string)`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemental function.
+
+#### Argument
+
+- `string`: Instance of a `string_type`. This argument is `intent(in)`.
+
+#### Result value
+
+The result is a default integer scalar value.
+
+#### Example
+
+```fortran
+use stdlib_string_type
+implicit none
+type(string_type) :: string
+integer :: code
+
+string = "Fortran"
+code = iachar(string)
+end
+```
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Index function
+
+Position of a *substring* within a *string*.
+
+#### Description
+
+Returns the position of the start of the leftmost or rightmost occurrence
+of string *substring* in *string*, counting from one. If *substring* is not
+present in *string*, zero is returned.
+
+#### Syntax
+
+`res = [[stdlib_string_type(module):index(interface)]] (string, substring[, back])`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemental function.
+
+#### Argument
+
+- `string`: Either scalar character value or string type. This argument is `intent(in)`.
+- `substring`: Either scalar character value or string type. This argument is `intent(in)`.
+- `back`: Either absent or a scalar logical value. This argument is `intent(in)`.
+
+#### Result value
+
+The result is a default integer scalar value.
+
+#### Example
+
+```fortran
+use stdlib_string_type
+implicit none
+type(string_type) :: string
+integer :: pos
+
+string = "Search this string for this expression"
+pos = index(string, "this")
+! pos == 8
+
+pos = index(string, "this", back=.true.)
+! pos == 24
+
+pos = index(string, "This")
+! pos == 0
+end
+```
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Scan function
+
+Scan a *string* for the presence of a *set* of characters. Scans a *string* for
+any of the characters in a *set* of characters.
```

Comment:
The two sentences are very similar. Is it written similarly in the documentation of the intrinsic function?

Reply:
Most descriptions are oriented at the descriptions of the intrinsic functions, since they are in fact implementing supplements to the intrinsics.

Reply:
```suggestion
Scan a *string* for the presence of a *set* of characters. 
```
Personally, I would drop the second sentence. I think the fact the dummy argument is called `set` and not `substring` is clear enough to indicate that we are scanning for *any* of the characters.